compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3827,6 +3830,48 @@ private boolean evaluateRawTemplateExprs(List<? extends BLangExpression> exprs,
         return errored;
     }
 
+    private BType getResolvedIntersectionType(BType type) {
+        return type.tag != TypeTags.INTERSECTION ? type : ((BIntersectionType) type).effectiveType;
+    }
+
+    private boolean containsAnyType(BType type) {
+        if (type == symTable.anyType) {
+            return true;
+        }
+
+        if (type.tag == TypeTags.UNION) {
+            return ((BUnionType) type).getMemberTypes().contains(symTable.anyType);
+        }
+
+        return false;
+    }
+
+    private BType getCompatibleRawTemplateType(BType expType, DiagnosticPos pos) {
+        if (expType.tag != TypeTags.UNION) {
+            return expType;
+        }
+
+        BUnionType unionType = (BUnionType) expType;
+        List<BType> compatibleTypes = new ArrayList<>();
+
+        for (BType type : unionType.getMemberTypes()) {
```

Comment:
Should we call `getEffectiveMemberTypes` here instead of `getMemberTypes` ?
https://github.com/ballerina-platform/ballerina-lang/blob/1cbfd04db73a9f36a46bdb47e8298bc5c4f39374/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L2336

This is currently used in 
https://github.com/ballerina-platform/ballerina-lang/blob/1cbfd04db73a9f36a46bdb47e8298bc5c4f39374/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L2297

Reply:
I guess the `isAssignable` check will handle this for us.

Reply:
This is just to iterate through the union members to see if there are compatible raw template types. No need to call `getEffectiveMemberTypes()` for that right?

Reply:
👍